cds/modules/deposit/api.py

```diff
@@ -787,6 +793,39 @@ def _tasks_global_status(self):
         iterate_events_results(events=events, fun=global_status)
         return global_status.status
 
+    def _rename_subtitles(self):
+        pattern = re.compile('(?P<filename>.*)_(?P<rest>[a-zA-Z]{2}\.vtt$)')
+        subtitles = CDSVideosFilesIterator.get_video_subtitles(self)
+        for subtitle_file in subtitles:
+            subtitle_obj = as_object_version(subtitle_file['version_id'])
+            match = pattern.match(subtitle_file['key'])
+            if match:
+                subtitle_obj.key = '{}_{}'.format(self['report_number'][0],
+                                                  match.group('rest'))
+                db.session.add(subtitle_obj)
+
+    def _publish_new(self, id_=None):
+        """Rename master file and subtitles before first publishing."""
```

Comment:
If I understood correctly, this function is renaming but also publishing at the same time? Could you, please, update the docstring to something like "Rename master file and subtitles **and publish for the first time**" or something similar ?

Reply:
Yes, or what about splitting this code in:

```
        id_ = id_ or uuid.uuid4()
        self.mint_report_number(id_)
        self['_files'] = self.files.dumps()

        master_file = CDSVideosFilesIterator.get_master_video_file(self)
        
        if master_file:
                self._rename_master(master_file)
                self._rename_subtitles()

        return super(CDSDeposit, self)._publish_new(id_)
```